src/DeviceInput/InputDevices/deviceSourceManager.ts

```diff
@@ -192,10 +195,12 @@ export class DeviceSourceManager implements IDisposable {
             case DeviceType.Generic:
                 const devices = this._devices[type];
                 delete this._firstDevice[type];

```

Comment:
nit: logically speaking it would make more sense if these lines were reversed
```suggestion
                delete this._firstDevice[type];
                const devices = this._devices[type];
```